tests/NLog.Web.Tests/NLogHttpModuleTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Linq; 
+using System.Web;
+using NLog.Web.Targets.Wrappers;
+using NLog.Web.Tests.LayoutRenderers;
+using Xunit;
+
+namespace NLog.Web.Tests
+{
+    public class NLogHttpModuleTests : TestInvolvingAspNetHttpContext
+    {
+        private static LogFactory RegisterAspNetBufferingTargetWrapper(string targetName)
+        {
+            LogManager.LogFactory.Setup().RegisterNLogWeb().LoadConfigurationFromXml(
+                $@"<nlog throwConfigExceptions='true'>
+                    <targets>
+                        <target 
+                            type='AspNetBufferingWrapper' 
+                            name='{targetName}' 
+                            bufferGrowLimit='9' 
+                            bufferSize='7' 
+                            growBufferAsNeeded='true'>
+                            <target type='Memory' name='unitTestMemoryTarget' />
+                        </target>
+                    </targets>
+                    <rules>
+                        <logger name='*' minlevel='Debug' writeTo='{targetName}' />
+                    </rules>
+                </nlog>");
+            return LogManager.LogFactory;
+        }
+
+        [Fact]
+        public void SuccessTest()
+        {
+            // Arrange
+            var httpContext = SetUpFakeHttpContext();
+
+            var logFactory = RegisterAspNetBufferingTargetWrapper("first");
+
+            Assert.NotNull(logFactory?.Configuration?.FindTargetByName<AspNetBufferingTargetWrapper>("first"));
+
+            var target = logFactory?.Configuration?.FindTargetByName<AspNetBufferingTargetWrapper>("first");
+            target.HttpContextAccessor = new FakeHttpContextAccessor(new HttpContextWrapper(httpContext));
+
+            // Act
+            var httpModule = new NLogHttpModule();
+            httpModule.OnBeginRequest(httpContext);
+
+            ILogger logger = logFactory.GetCurrentClassLogger();
+
+            for (int i = 0; i < 10; i++)
+            {
+                logger.Debug("This is a unit test logging.");
+            }
+
+            // Assert
+            Assert.NotNull(httpContext.Items);
```

Comment:
Instead of doing white-box Assert of the inner-details inside `httpContext.Item`, then I think the unit-test should focus on checking `unitTestMemoryTarget`:
- Logging without having active HttpContext will make it forward directly to `unitTestMemoryTarget`
- Logging within active HttpContext will not reach `unitTestMemoryTarget` until after `OnEndRequest`